virtualbox/vbox-clean-snapshots.py

```diff
@@ -3,49 +3,77 @@
 import sys
 import argparse
 import textwrap
-import virtualbox
-from virtualbox.library import MachineState
-
-
-TO_DELETE = []
-
-
-def get_snapshots_to_delete(snapshot, protected_snapshots):
-    for child in snapshot.children:
-        get_snapshots_to_delete(child, protected_snapshots)
-    snapshot_name = snapshot.name.lower()
-    for protected_str in protected_snapshots:
-        if protected_str.lower() in snapshot_name:
-            return
-    TO_DELETE.append((snapshot.name, snapshot.id_p))
-
+import subprocess
+import re
+from vboxcommon import *
+
+def get_snapshot_children(vm_name, root_snapshot_name, protected_snapshots):
+    """Gets the children of a snapshot using 'VBoxManage showvminfo'.
+    
+    Args:
+      vm_name: The name of the VM.
+      snapshot_name: The name of the snapshot.
+
+    Returns:
+      A list of snapshot names that are children of the given snapshot.
```

Comment:
nice documentation!

Reply:
does this return all descendents recursively? or only the direct children?

Reply:
It should return all. I think it would be good to clarify it in the documentation. :+1: